### ISSUE TITLE
EcsRunLauncher regression - "Could not determine image for run"

### ISSUE THREAD

**Issue by treilly1** [dagster-io/dagster#26138]
### What's the issue?

* detailed outline in dagster slack https://dagster.slack.com/archives/C01U954MEER/p1732566748483749
* Attempting an upgrade to dagster `1.9.2` raises error [here](https://github.com/dagster-io/dagster/blob/94bcd2a42a9f37f65613e21ee8b5704ef94d6204/python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py#L490-L491) despite my `dagster.yaml` configuring the run launcher as outlined in [current documentation](https://docs.dagster.io/deployment/guides/aws#launching-runs-in-ecs) 

### What did you expect to happen?

* No breaking `ECSRunLauncher` changes when upgrading. Since underlying image is specified on my ECS task def (and controlled via CICD process), I'd expect dagster to use image defined on task def. 

### How to reproduce?

* For ECS launchers configured like documentation outlines, attempt kicking off a run.
```
run_launcher:
  module: "dagster_aws.ecs"
  class: "EcsRunLauncher"
  config:
    task_definition: "arn:aws:ecs:us-east-1:1234567890:task-definition/my-task-definition:1"
    container_name: "my_container_name"
```

### Dagster version

`1.9.2`

### Deployment type

Other Docker-based deployment

### Deployment details

* All dagster resources are deployed to an ECS cluster

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.
By submitting this issue, you agree to follow Dagster's [Code of Conduct](https://github.com/dagster-io/dagster/blob/master/.github/CODE_OF_CONDUCT.md).

**Comment by zyd14** [comment]
I have a PR to roll back the check I added that is causing the error; but tracing through the code it's unclear to me that this will actually fix the issue. It seems like `image` is being pulled from the [job's repository_origin](https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py#L431-L433), and it seems like `run_task_kwargs` needs [image to be non-None ](https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py#L652-L678). There's a lot of infra to set up to be able to test this, I'm hoping @treilly1 will be able to test my changes with their existing infra if possible

**Comment by lokinnnnnn** [comment]
I also encountered this issue, so I returned to version 1.9.0

**Comment by treilly1** [comment]
I added `DAGSTER_CURRENT_IMAGE` to my gRPC task environment variables and then `1.9.2` worked for me. It seems like the value of `DAGSTER_CURRENT_IMAGE` doesn't matter it just has to exist. I set mine to `"PLACEHOLDER"` and confirmed the following behavior:
1. My run task starts up by invoking dagster internal command `dagster api execute_run` and one of the arguments is now `"container_image": "PLACEHOLDER"`
2. In the UI, my code locations now specify the `image` metadata 
![image](https://github.com/user-attachments/assets/ced9e41c-c43d-4405-90ce-bf9bc4bd99f3)
3. My run tasks **do not override their image** with `PLACEHOLDER` despite two points above.

This was discussed a bit more detailed in dagster slack [thread](https://dagster.slack.com/archives/C01U954MEER/p1732566748483749), but the intended behavior isn't clear to me. Specifically, on whether or not `DAGSTER_CURRENT_IMAGE` is supposed to be an override. The behavior I witnessed seems to indicate it is read only metadata now required to use ECS in `1.9.2`. 

If someone from dagster could chime in on the desired behavior, I would consider contributing to a solution.
